### DISCUSSION TITLE
App Hooks in components?

### DISCUSSION THREAD

**Discussion by mercs600** [nuxt/nuxt#23778] [Questions]
In the documentation, we can read that we can use app hooks in plugins as well as in composables. 
What about components? Can we register and use custom app hooks in components like this?

```js
const { hooks } = useNuxtApp()
const unregisterDoSomething = hooks.hook('app:custom-hook', doSomething)
```
Is it safe when it comes to memory management? 

Can we register them in setup? or only in client side context to be able unregister them on destroy? 

How about unregistering the hook when the component is destroyed?
should we unregister them this way? 
```js
unregisterDoSomething()
// or 
hooks.removeHook('app:custom-hook', doSomething)
```

**Comment by huang-julien** [comment]
Hello :wave: yes you should definitively unregister the hook, otherwise you'll have a memory leak. 
You can use the unregister function provided by `.hook()`
